x-pack/plugins/security_solution/cypress/tasks/api_calls/rules.ts

```diff
@@ -51,98 +55,107 @@ export const createCustomRule = (rule: CustomRule, ruleId = 'rule_testing', inte
 
-export const createEventCorrelationRule = (rule: CustomRule, ruleId = 'rule_testing') =>
-  cy.request({
-    method: 'POST',
-    url: 'api/detection_engine/rules',
-    body: {
-      rule_id: ruleId,
-      risk_score: parseInt(rule.riskScore, 10),
-      description: rule.description,
-      interval: `${rule.runsEvery.interval}${rule.runsEvery.type}`,
-      from: `now-${rule.lookBack.interval}${rule.lookBack.type}`,
-      name: rule.name,
-      severity: rule.severity.toLocaleLowerCase(),
-      type: 'eql',
-      index: rule.index,
-      query: rule.customQuery,
-      language: 'eql',
-      enabled: true,
-    },
-    headers: { 'kbn-xsrf': 'cypress-creds' },
-  });
+export const createEventCorrelationRule = (rule: CustomRule, ruleId = 'rule_testing') => {
+  if (rule.dataSource.type === 'indexPatterns') {
+    cy.request({
+      method: 'POST',
+      url: 'api/detection_engine/rules',
+      body: {
+        rule_id: ruleId,
+        risk_score: parseInt(rule.riskScore, 10),
+        description: rule.description,
+        interval: `${rule.runsEvery.interval}${rule.runsEvery.type}`,
+        from: `now-${rule.lookBack.interval}${rule.lookBack.type}`,
+        name: rule.name,
+        severity: rule.severity.toLocaleLowerCase(),
+        type: 'eql',
+        index: rule.dataSource.index,
+        query: rule.customQuery,
+        language: 'eql',
+        enabled: true,
+      },
+      headers: { 'kbn-xsrf': 'cypress-creds' },
+    });
+  }
+};
 
-export const createCustomIndicatorRule = (rule: ThreatIndicatorRule, ruleId = 'rule_testing') =>
-  cy.request({
-    method: 'POST',
-    url: 'api/detection_engine/rules',
-    body: {
-      rule_id: ruleId,
-      risk_score: parseInt(rule.riskScore, 10),
-      description: rule.description,
-      // Default interval is 1m, our tests config overwrite this to 1s
-      // See https://github.com/elastic/kibana/pull/125396 for details
-      interval: '10s',
-      name: rule.name,
-      severity: rule.severity.toLocaleLowerCase(),
-      type: 'threat_match',
-      timeline_id: rule.timeline.templateTimelineId,
-      timeline_title: rule.timeline.title,
-      threat_mapping: [
-        {
-          entries: [
-            {
-              field: rule.indicatorMappingField,
-              type: 'mapping',
-              value: rule.indicatorIndexField,
-            },
-          ],
-        },
-      ],
-      threat_query: '*:*',
-      threat_language: 'kuery',
-      threat_filters: [],
-      threat_index: rule.indicatorIndexPattern,
-      threat_indicator_path: rule.threatIndicatorPath,
-      from: 'now-50000h',
-      index: rule.index,
-      query: rule.customQuery || '*:*',
-      language: 'kuery',
-      enabled: true,
-    },
-    headers: { 'kbn-xsrf': 'cypress-creds' },
-    failOnStatusCode: false,
-  });
+export const createCustomIndicatorRule = (rule: ThreatIndicatorRule, ruleId = 'rule_testing') => {
+  if (rule.dataSource.type === 'indexPatterns') {
```

Comment:
Instead of having this condition, let's support both index patterns and data views here as in `createCustomRule`.

Reply:
Same as above.